Mathlib/Order/Hom/Lex.lean

```diff
@@ -0,0 +1,177 @@
+/-
+Copyright (c) 2025 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Data.Sum.Order
+import Mathlib.Order.Hom.Set
+import Mathlib.Order.RelIso.Set
+
+/-!
+# Lexicographic order and order isomorphisms
+
+The main result in this file is the following: if `α` is a linear order and `x : α`, then `α` is
+order isomorphic to either of `Iio x ⊕ₗ Ici x` or `Iic x ⊕ₗ Ioi x`.
+-/
+
+open Set
+
+variable {α : Type*}
+
+/-! ### Relation isomorphism -/
+
+namespace RelIso
+
+variable {r : α → α → Prop} {x y : α} [IsTrans α r] [IsTrichotomous α r] [DecidableRel r]
+
+variable (r x) in
+/-- A relation is isomorphic to the lexicographic sum of elements less than `x` and elements not
+less than `x`. -/
+-- The explicit type signature stops `simp` from complaining.
+def sumLexComplLeft :
+    @Sum.Lex {y // r y x} {y // ¬ r y x} (Subrel r {y | r y x}) (Subrel r {y | ¬ r y x}) ≃r r where
+  toEquiv := Equiv.sumCompl (r · x)
+  map_rel_iff' := by
+    rintro (⟨a, ha⟩ | ⟨a, ha⟩) (⟨b, hb⟩ | ⟨b, hb⟩)
+    · simp
+    · simpa using trans_trichotomous_right ha hb
+    · simpa using fun h ↦ ha <| trans h hb
+    · simp
+
+@[simp]
+theorem sumLexComplLeft_apply_inl (a : {y // r y x}) : sumLexComplLeft r x (Sum.inl a) = a :=
+  rfl
+
+@[simp]
+theorem sumLexComplLeft_apply_inr (a : {y // ¬ r y x}) : sumLexComplLeft r x (Sum.inr a) = a :=
+  rfl
+
+theorem sumLexComplLeft_symm_apply_of_mem (h : r y x) :
+    (sumLexComplLeft r x).symm y = Sum.inl ⟨y, h⟩ :=
+  Equiv.sumCompl_apply_symm_of_pos (r · x) _ h
+
+theorem sumLexComplLeft_symm_apply_of_not_mem (h : ¬ r y x) :
+    (sumLexComplLeft r x).symm y = Sum.inr ⟨y, h⟩ :=
+  Equiv.sumCompl_apply_symm_of_neg (r · x) _ h
+
+@[simp]
+theorem sumLexComplLeft_symm_apply (a : {y // r y x}) :
+    (sumLexComplLeft r x).symm a = Sum.inl a :=
+  sumLexComplLeft_symm_apply_of_mem a.2
+
+@[simp]
+theorem sumLexComplLeft_symm_apply_neg (a : {y // ¬ r y x}) :
+    (sumLexComplLeft r x).symm a = Sum.inr a :=
+  sumLexComplLeft_symm_apply_of_not_mem a.2
```

Comment:
I would suggest dropping all of these and just having the lemmas
* `sumLexComplLeft r x y = Equiv.sumCompl (r · x) y`
* `(sumLexComplLeft r x).symm y = (Equiv.sumCompl (r · x)).symm y`

and letting the API around `sumCompl` handle it.